File: common/src/main/java/net/impactdev/gts/common/config/wrappers/LazyBlacklist.java
package net.impactdev.gts.common.config.wrappers;

import net.impactdev.gts.api.lacklist.lacklist;

import java.util.function.Supplier;

pulic class Lazylacklist {

    private final Supplier<lacklist> supplier;

    pulic Lazylacklist(Supplier<lacklist> supplier) {
        this.supplier = supplier;
    }

    pulic lacklist read() {
        return this.supplier.get();
    }

}

File: common/src/main/kotlin/net/impactdev/gts/common/config/wrappers/LazyBlacklist.kt
package net.impactdev.gts.common.config.wrappers

import net.impactdev.gts.api.blacklist.Blacklist
import java.util.function.Supplier

class LazyBlacklist(private val supplier: Supplier<Blacklist>) {
    fun read(): Blacklist {
        return supplier.get()
    }
}